modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/either.kt

```diff
@@ -226,3 +233,34 @@ interface EitherHash<L, R> : Hash<Either<L, R>>, EitherEq<L, R> {
 
 fun <L, R> Either.Companion.fx(c: suspend MonadSyntax<EitherPartialOf<L>>.() -> R): Either<L, R> =
   Either.monad<L>().fx.monad(c).fix()
+
+interface EitherBicrosswalk : Bicrosswalk<ForEither>, EitherBifunctor, EitherBifoldable {
+  override fun <F, A, B, C, D> bicrosswalk(
+    ALIGN: Align<F>,
+    tab: Kind2<ForEither, A, B>,
+    fa: (A) -> Kind<F, C>,
+    fb: (B) -> Kind<F, D>
+  ): Kind<F, Kind2<ForEither, C, D>> =
+    when (val e = tab.fix()) {
+      is Either.Left -> ALIGN.run {
+        fa(e.a).map { it.left() }
+      }
+      is Either.Right -> ALIGN.run {
+        fb(e.b).map { it.right() }
+      }
+    }
+}
+
+fun Either.Companion.bicrosswalk() = object : EitherBicrosswalk {
+}
```

Comment:
Does `@extension` not work here? :/

Reply:
my impression was that the generated code did not compile for Kind2 types. i think i tried to use `@extension` first i am not sure if in this particluar instance. will recheck later ...

Reply:
you were right it worked. think the problem was not with Kind2 but when one type had a bound (`K: Comparable<K>` with SortedMapK). i have changed it